cmd/manager/aggregator.go

```diff
@@ -336,6 +337,31 @@ func createOrUpdateOneResult(crClient aggregatorCrClient, owner metav1.Object, l
 	return nil
 }
 
+func setTimestampAnnotations(crClient aggregatorCrClient, owner metav1.Object, annotations map[string]string) map[string]string {
+	// Get the scan timestamp
+	scan := &compv1alpha1.ComplianceScan{}
+	err := backoff.Retry(func() error {
+		err := crClient.getClient().Get(context.TODO(), types.NamespacedName{
+			Namespace: owner.GetNamespace(),
+			Name:      owner.GetName(),
+		}, scan)
+		if err != nil {
+			return err
+		}
+		return nil
+	}, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), maxRetriesForTimestamp))
+	if err != nil {
+		cmdLog.Error(err, "Failed to get scan", "ComplianceScan.Name", owner.GetName())
+		return annotations
```

Comment:
Won't this result in a nil pointer exception if we don't have any annotations passed in? Should lines 357 - 359 be moved to the top of the function so we guarantee we always return a map of strings, even if we can't get the scan?

Reply:
This should be addressed in https://github.com/ComplianceAsCode/compliance-operator/pull/591/files#diff-37d868832c42510177fe527ffee41dc25b205168859c7a1467e2b9f592fb4ed3L531